openedx/core/djangoapps/announcements/views.py

```diff
@@ -0,0 +1,100 @@
+"""
+Views to show announcements.
+"""
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.template.loader import render_to_string
+from django.utils.translation import ugettext as _
+from web_fragments.fragment import Fragment
+
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+
+from .models import Announcements
+
+
+class AnnouncementsView(EdxFragmentView):
+    """
+    View showing a list of all announcements.
+    """
+    template = "announcements/announcements_list.html"
+
+    def render_to_fragment(self, request):
+        """
+        Render page which lists all announcements
+        """
+        announcements = Announcements.objects.all()
+        context = {
+            #"announcement_list": []
+            "announcement_list": announcements
+        }
+        html = render_to_string(self.template, context, request=request)
+        return Fragment(html)
+
+
+class AnnouncementsEditView(EdxFragmentView):
```

Comment:
I think adding/editing announcements should only be allowed to a limited number of users, say only users with `is_staff` or "course creator" status?  see e.g. `cms.djangoapps.contentstore.views/course._get_course_creator_status`, though this is a terrible place for this method to be if we want to check it from `openedx`.

Reply:
Ah, I mistakenly thought studio login was restricted to staff already.
I've used require_global_staff from util.views to fix this.